superduperdb/container/artifact.py

```diff
@@ -44,14 +44,36 @@ def __init__(
         serializer: str = 'dill',
         sha1: str = '',
         hash: t.Optional[int] = None,
+        artifact_store: t.Optional['ArtifactStore'] = None
     ):
-        self.artifact = artifact
+        self._artifact = artifact
         self.file_id = file_id
         self.info = info
         self.object_id = object_id
         self.serializer = serializer
         self.hash = hash
         self._sha1 = sha1
+        self.artifact_store = artifact_store
+
+    @property
+    def artifact(self):
```

Comment:
also we can get rid of _artifact with 
cached property
wdyt

Reply:
Right.

Reply:
No, actually this won't work because of the `__init__` method.